agents/agents-features/agents-features-snapshot/src/commonMain/kotlin/ai/koog/agents/snapshot/providers/filters/AgentCheckpointFilter.kt

```diff
@@ -0,0 +1,27 @@
+package ai.koog.agents.snapshot.providers.filters
+
+import ai.koog.agents.snapshot.feature.AgentCheckpointData
+
+/**
+ * Interface for filtering agent checkpoints.
+ */
+public interface AgentCheckpointFilter
```

Comment:
Why do we need two separate interfaces here? Why can't `AgentCheckpointFilter` have `check` method? What was the idea behind marker interface?